src/MonoTorrent/MonoTorrent.Client.Messages.Libtorrent/LTMetadata.cs

```diff
@@ -91,15 +82,15 @@ public LTMetadata (byte extensionId, eMessageType type, int piece, byte[] metada
             : this ()
         {
             ExtensionId = extensionId;
-            this.messageType = type;
-            this.metadata = metadata;
-            this.piece = piece;
+            this.MetadataMessageType = type;
```

Comment:
the `this.` is unnecessary now. Can a refactoring remove those, and a formatter setting ensure they stay removed?

Reply:
There are cases when a parameter name equals to field name so this. can not be removed. You can convert private field names from camelCase to _camelCase format in resharper.